### PR TITLE
feat: show debug mode in UI

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes NAN-101

- Show if debug mode is activated in the UI
- Move admin action into it's own component
- POST /admin/switch now returns proper json error to display in the UI
- Debug mode expires after 10minutes 

## How to test
- In `account.controller.ts` modify `AUTH_ENABLED=true` and `AUTH_ADMIN_SWITCH_ENABLED=true`
- In your `.env` add your local `NANGO_ADMIN_UUID`
- Test the feature

## Results

I tried something that's easy to read, hard to miss, but easy to not capture in screenshots, let me know.
Screenshot 2024-03-08 at 15 31 03

### CODE REVIEW

File: packages/server/lib/controllers/account.controller.ts [hunk 1]
```diff
@@ -128,15 +137,13 @@ class AccountController {
...
-                //environment_id: result.environmentId
                 environment_id: response.environment.id
             };
 
             await createActivityLogAndLogMessage(log, {
                 level: 'info',
                 environment_id: response.environment.id,
                 timestamp: Date.now(),
-                //content: `A Nango admin logged into your account for the following reason: "${login_reason}"`
                 content: `A Nango admin logged into another account for the following reason: "${login_reason}"`
```
Comment (khaliqgant) [inline packages/server/lib/controllers/account.controller.ts:147]: I still think we don't want to show this messaging which is why it was commented out. So you can delete this completely.

File: packages/server/lib/server.ts [hunk 2]
```diff
@@ -142,7 +142,7 @@ app.route('/admin/flow/deploy/pre-built').post(apiAuth, flowController.adminDepl
 app.route('/proxy/*').all(apiAuth, upload.any(), proxyController.routeCall.bind(proxyController));
 
 // Webapp routes (no auth).
-if (isCloud() || isEnterprise()) {
+if (AUTH_ADMIN_SWITCH_ENABLED) {
```
Comment (khaliqgant) [inline packages/server/lib/server.ts]: This variable is incorrectly named for this context

File: packages/webapp/src/components/DebugMode.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { useStore } from '../store';
+
```
Comment (khaliqgant) [inline packages/webapp/src/components/DebugMode.tsx:2]: nice abstraction

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Looks good, changes requested inline

File: packages/server/lib/controllers/account.controller.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
 import type { Request, Response, NextFunction } from 'express';
-import { isCloud, accountService, userService, errorManager, LogLevel, LogActionEnum, createActivityLogAndLogMessage } from '@nangohq/shared';
+import { accountService, userService, errorManager, LogLevel, LogActionEnum, createActivityLogAndLogMessage, isEnterprise, isCloud } from '@nangohq/shared';
 import { getUserAccountAndEnvironmentFromSession } from '../utils/utils.js';
 
+export const AUTH_ENABLED = isCloud() || isEnterprise();
+export const NANGO_ADMIN_UUID = process.env['NANGO_ADMIN_UUID'];
+export const AUTH_ADMIN_SWITCH_ENABLED = NANGO_ADMIN_UUID && isCloud();
+export const AUTH_ADMIN_SWITCH_MS = 600 * 1000;
```
Comment (TBonnin) [inline packages/server/lib/controllers/account.controller.ts:8]: ❤️

**Review by TBonnin — APPROVED** [review]

**Review by TBonnin — COMMENTED** [review]
love the ttl

File: packages/server/lib/server.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/server/lib/server.ts]: oops yes wrong c/c thanks

File: packages/server/lib/controllers/account.controller.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/server/lib/controllers/account.controller.ts:147]: For compliance it's often required to log something, maybe @bastienbeurier have an opinion on this?

Comment (khaliqgant) [reply packages/server/lib/controllers/account.controller.ts:147]: I wanted to have this messaging here as well, which is why I commented it out and didn't delete it FWIW

Comment (bastienbeurier) [reply packages/server/lib/controllers/account.controller.ts:147]: I agree we should log this at some point. I don't mind whether it's now or later. I slightly prefer waiting for our project on logs DX where we'll redefine everything that needs to be logged.

Comment (khaliqgant) [reply packages/server/lib/controllers/account.controller.ts:147]: It still should be “your” account if we do show it

Comment (bodinsamuel) [reply packages/server/lib/controllers/account.controller.ts:147]: alright let's not touch this and rediscuss ☺️

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**